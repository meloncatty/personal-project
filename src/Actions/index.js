import {apiKey} from '../apiKey.js'

export const resultsSuccess = (results) => ({
  type: 'QUERY_RESULTS_SUCCESS',
  results
})

export const resultsAreLoading = (bool) => ({
  type: 'RESULTS_ARE_LOADING',
  resultsLoading: bool
})

export const resultsHaveErrored = (bool) => ({
  type: 'RESULTS_HAVE_ERRORED',
  resultsErrored: bool
})

export const fullArticleLoading = (bool) => ({
  type: 'FULL_ARTICLE_LOADING',
  articleLoading: bool
})

export const fullArticleErrored = (bool) => ({
  type: 'FULL_ARTICLE_ERRORED',
  articleErrored: bool
})

export const fullArticleSuccess = (result) => ({
  type: 'FULL_ARTICLE_SUCCESS',
  result
})

export const userAuthentication = (userData) => ({
  type: 'USER_AUTHENTICATION',
  userAuth: userData
})

export const isUserSignedIn = (bool) => ({
  type: 'IS_USER_SIGNED_IN',
  isUserSignedIn: bool
})

export const captureUserArticles = (articles) => ({
  type: 'CAPTURE_USER_ARTICLES',
  userArticles: articles
})

export const fetchArticles = (query) => {
  return async (dispatch) => {
    try {
      dispatch(resultsSuccess([]))
      dispatch(resultsHaveErrored(false))
      dispatch(resultsAreLoading(true))
      const url = `https://core.ac.uk:443/api-v2/articles/search/${query}?page=1&pageSize=10&metadata=true&fulltext=false&citations=true&similar=false&duplicate=false&urls=true&faithfulMetadata=false&apiKey=${apiKey}`
      const response = await fetch(url)
      if(!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(resultsAreLoading(false))
      const articles = await response.json()
      dispatch(resultsSuccess(articles.data))
    } catch (error) {
      dispatch(resultsHaveErrored(true))
      dispatch(resultsAreLoading(false))
    }
  }
}

export const fetchFullText = (id) => {
  return async (dispatch) => {
    try {
      dispatch(fullArticleLoading(true))
      const url = `https://core.ac.uk:443/api-v2/articles/get/${id}?metadata=true&fulltext=true&citations=false&similar=false&duplicate=false&urls=false&faithfulMetadata=false&apiKey=${apiKey}`
      const response = await fetch(url)
      if(!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(fullArticleLoading(false))
      const article = await response.json()
      dispatch(fullArticleSuccess(article.data))
    } catch (error) {
      dispatch(fullArticleErrored(true))
      dispatch(fullArticleLoading(false))
    }
  }
}
