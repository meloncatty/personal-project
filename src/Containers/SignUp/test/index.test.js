import React from 'react'
import {shallow} from 'enzyme'
import {SignUp, mapDispatchToProps} from '../'
import {userSignupSuccess} from '../../../Actions'

describe('SignUp', () => {
  let signUp
  
  beforeEach(() => {
    signUp = shallow(<SignUp userSignupSuccess={jest.fn()} />)
  })

  it('should match snapshot', () => {
    expect(signUp).toMatchSnapshot()
  })

  it('should have default state', () => {
    const expected = {
      username: '',
      email: '',
      passwordAuth: '',
      password: '',
      error: null
    }

    expect(signUp.state()).toEqual(expected)
  })

  describe('handleSignUpChange', () => {
    it('should update username on change', () => {
      const mockEvent = {target: {name: 'username', value: 'username'}}

      signUp.find('.username').simulate('change', mockEvent)

      expect(signUp.state('username')).toEqual(mockEvent.target.value)
    })

    it('should update email on change', () => {
      const mockEvent = {target: {name: 'new-email', value: 'new-email'}}

      signUp.find('.new-email').simulate('change', mockEvent)

      expect(signUp.state('new-email')).toEqual(mockEvent.target.value)
    })

    it('should update passwordAuth on change', () => {
      const mockEvent = {target: {name: 'passwordAuth', value: 'password'}}

      signUp.find('.password-auth').simulate('change', mockEvent)

      expect(signUp.state('passwordAuth')).toEqual(mockEvent.target.value)
    })

    it('should update password on change', () => {
      const mockEvent = {target: {name: 'password', value: 'password'}}

      signUp.find('.password-sign-up').simulate('change', mockEvent)

      expect(signUp.state('password')).toEqual(mockEvent.target.value)
    })
  })

  describe('handleSignUpSubmit', () => {
    it.skip('should call userSignupSuccess with true', () => {
      const mockEvent = {preventDefault: () => {}}
      signUp.find('form').simulate('submit', mockEvent)

      expect(signUp.instance().props.userSignupSuccess).toHaveBeenCalledWith(true)
    })

    it.skip('should call history.push', () => {
      const mockEvent = {preventDefault: () => {}}
      signUp.find('form').simulate('submit', mockEvent)

      expect(signUp.instance().props.history).toHaveBeenCalled()
    })
  })

  describe('mapDispatchToProps', () => {
    it('should return false when user signs out', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = userSignupSuccess(true)
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.userSignupSuccess(true)

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})
