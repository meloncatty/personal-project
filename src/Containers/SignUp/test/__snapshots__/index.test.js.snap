// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUp
    userSignupSuccess={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="sign-up-form"
        onSubmit={[Function]}
>
        <label>
                Create An Account
        </label>
        <input
                className="username"
                name="username"
                onChange={[Function]}
                placeholder="Full Name"
                type="text"
        />
        <input
                className="new-email"
                name="email"
                onChange={[Function]}
                placeholder="Email Address"
                type="text"
        />
        <input
                className="password-sign-up"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
        />
        <input
                className="password-auth"
                name="passwordAuth"
                onChange={[Function]}
                placeholder="Confirm Password"
                type="password"
        />
        <button
                disabled={true}
                type="submit"
        >
                Sign Up
        </button>
</form>,
      "className": "form-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label>
            Create An Account
</label>,
          <input
            className="username"
            name="username"
            onChange={[Function]}
            placeholder="Full Name"
            type="text"
/>,
          <input
            className="new-email"
            name="email"
            onChange={[Function]}
            placeholder="Email Address"
            type="text"
/>,
          <input
            className="password-sign-up"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
/>,
          <input
            className="password-auth"
            name="passwordAuth"
            onChange={[Function]}
            placeholder="Confirm Password"
            type="password"
/>,
          <button
            disabled={true}
            type="submit"
>
            Sign Up
</button>,
          null,
        ],
        "className": "sign-up-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Create An Account",
          },
          "ref": null,
          "rendered": "Create An Account",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "username",
            "name": "username",
            "onChange": [Function],
            "placeholder": "Full Name",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "new-email",
            "name": "email",
            "onChange": [Function],
            "placeholder": "Email Address",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "password-sign-up",
            "name": "password",
            "onChange": [Function],
            "placeholder": "Password",
            "type": "password",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "password-auth",
            "name": "passwordAuth",
            "onChange": [Function],
            "placeholder": "Confirm Password",
            "type": "password",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Up",
            "disabled": true,
            "type": "submit",
          },
          "ref": null,
          "rendered": "Sign Up",
          "type": "button",
        },
        null,
      ],
      "type": "form",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="sign-up-form"
          onSubmit={[Function]}
>
          <label>
                    Create An Account
          </label>
          <input
                    className="username"
                    name="username"
                    onChange={[Function]}
                    placeholder="Full Name"
                    type="text"
          />
          <input
                    className="new-email"
                    name="email"
                    onChange={[Function]}
                    placeholder="Email Address"
                    type="text"
          />
          <input
                    className="password-sign-up"
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
          />
          <input
                    className="password-auth"
                    name="passwordAuth"
                    onChange={[Function]}
                    placeholder="Confirm Password"
                    type="password"
          />
          <button
                    disabled={true}
                    type="submit"
          >
                    Sign Up
          </button>
</form>,
        "className": "form-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Create An Account
</label>,
            <input
              className="username"
              name="username"
              onChange={[Function]}
              placeholder="Full Name"
              type="text"
/>,
            <input
              className="new-email"
              name="email"
              onChange={[Function]}
              placeholder="Email Address"
              type="text"
/>,
            <input
              className="password-sign-up"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
/>,
            <input
              className="password-auth"
              name="passwordAuth"
              onChange={[Function]}
              placeholder="Confirm Password"
              type="password"
/>,
            <button
              disabled={true}
              type="submit"
>
              Sign Up
</button>,
            null,
          ],
          "className": "sign-up-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Create An Account",
            },
            "ref": null,
            "rendered": "Create An Account",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "username",
              "name": "username",
              "onChange": [Function],
              "placeholder": "Full Name",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "new-email",
              "name": "email",
              "onChange": [Function],
              "placeholder": "Email Address",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "password-sign-up",
              "name": "password",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "password-auth",
              "name": "passwordAuth",
              "onChange": [Function],
              "placeholder": "Confirm Password",
              "type": "password",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign Up",
              "disabled": true,
              "type": "submit",
            },
            "ref": null,
            "rendered": "Sign Up",
            "type": "button",
          },
          null,
        ],
        "type": "form",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
